### PR TITLE
Fix part of #5070: Display empty answer message in Math expressions input interaction.

### PR DESCRIPTION
**Pull request by Vishwajith-Shettigar**
## Explanation
Fix part of #5070,  In   Math expressions Interaction UI, leave submit button enabled when answer is empty. Show an error on submitting an empty answer.

MathExpressionInteractionsViewTestActivity added to accessibility_label_exemptions and test_file_exemptions files.

### Numeric expression

[numerix_expression.webm](https://github.com/oppia/oppia-android/assets/76042077/ce7e3e2a-e209-4197-ac3b-4a4ab281ac53)

### Algebraic expression

[algebraic_expression.webm](https://github.com/oppia/oppia-android/assets/76042077/62222fe4-9cef-49cf-a86d-d74a0908d0a0)

### Math equation

[math_equation.webm](https://github.com/oppia/oppia-android/assets/76042077/fb097e88-c8b2-4159-bdcf-2b4e64386d78)

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by Vishwajith-Shettigar** [comment]
@adhiamboperes, PTAL.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt]: There is no guarantee someone seeing this file has seen the others so the comments are still important.
```suggestion
       inputAnswerAvailable = true // Allow blank answer submission.
```

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt:114]: Context is required here
` // Initializing with default values so that submit button is enabled by default.`

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt]: I this here you may be able to use `this` keyword to refer to the current class instead of passing an object of the class to its own functions.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt:448]: I don't see an else to the if branch.

File: app/src/main/res/layout/activity_math_expression_interaction_view_test.xml
Comment (adhiamboperes) [inline app/src/main/res/layout/activity_math_expression_interaction_view_test.xml]: Insert newline at EOF.

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (adhiamboperes) [inline app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: Update to MathExpressionInteractionViewTestActivityParams

File: app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt
Comment (adhiamboperes) [inline app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt:3705]: Here and everywhere else, why do we need to remove this?

File: domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt
Comment (adhiamboperes) [inline domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt:1]: Unnecessary formatting changes in file.

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @Vishwajith-Shettigar! 

I have some follow up comments.

File: app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt
Comment (Vishwajith-Shettigar) [reply app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt:3705]: Because now submit button is enabled after wrong answer submitted by the user.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (Vishwajith-Shettigar) [reply app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt:448]: I don't see any use of it.

File: app/src/main/res/layout/activity_math_expression_interaction_view_test.xml
Comment (Vishwajith-Shettigar) [reply app/src/main/res/layout/activity_math_expression_interaction_view_test.xml]: okay

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (Vishwajith-Shettigar) [reply app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt]: okay

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (Vishwajith-Shettigar) [reply app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt:114]: okay

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (Vishwajith-Shettigar) [reply app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt]: right.

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (Vishwajith-Shettigar) [reply app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: Done

File: domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt
Comment (Vishwajith-Shettigar) [reply domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressController.kt:1]: okay

**Comment by Vishwajith-Shettigar** [comment]
@adhiamboperes, PTAL.

**Comment by adhiamboperes** [comment]
@theMr17, could you please take a review pass on this PR? I had earlier requested some changes, please verify that they have been addressed, and please assign back to me once there is no comment to be addressed from your side.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (theMr17) [inline app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt]: Please add the argument names here, so that the code is self documented.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (theMr17) [inline app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt]: Nit: This trailing comma is redundant.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (theMr17) [inline app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt:1]: Nit: This file contains multiple redundant empty lines added, please remove them.

File: app/src/main/res/values/strings.xml
Comment (theMr17) [inline app/src/main/res/values/strings.xml:111]:
```suggestion
  <string name="math_expression_error_empty_input">Enter an equation to continue.</string>
```

File: app/src/main/res/layout/activity_input_interaction_view_test.xml
Comment (theMr17) [inline app/src/main/res/layout/activity_input_interaction_view_test.xml:1]: This file contains unnecessary code changes from Android Studio.

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (theMr17) [inline app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]:
```suggestion
  fun testView_allInteractions_blankInput_producesSubmitTimeError() {
```

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (theMr17) [inline app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]:
```suggestion
  fun testMathEquation_submitWithBlankInput_emptyInputErrorIsDisplayed() {
```

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (theMr17) [inline app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]:
```suggestion
  fun testAlgebraicExpression_submitWithBlankInput_emptyInputErrorIsDisplayed() {
```

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (theMr17) [inline app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]:
```suggestion
  fun testNumericExpression_submitWithBlankInput_emptyInputErrorIsDisplayed() {
```

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (theMr17) [inline app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: Espresso functions like `isDisplayed()`, `click()` & `withText()` should be directly imported. Make sure to change it for other added tests as well.

File: app/src/main/java/org/oppia/android/app/testing/MathExpressionInteractionsViewTestActivity.kt
Comment (theMr17) [inline app/src/main/java/org/oppia/android/app/testing/MathExpressionInteractionsViewTestActivity.kt]: Nit: Remove empty line.

**Review by theMr17 — CHANGES_REQUESTED** [review]
Thanks @Vishwajith-Shettigar! I have left some suggestions. Please make sure to complete the essential checklist as well.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (Vishwajith-Shettigar) [reply app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt:1]: Okay

File: app/src/main/res/values/strings.xml
Comment (Vishwajith-Shettigar) [reply app/src/main/res/values/strings.xml:111]: Looks good.

File: app/src/main/res/layout/activity_input_interaction_view_test.xml
Comment (Vishwajith-Shettigar) [reply app/src/main/res/layout/activity_input_interaction_view_test.xml:1]: I'll fix, thank you.

File: app/src/main/java/org/oppia/android/app/testing/MathExpressionInteractionsViewTestActivity.kt
Comment (Vishwajith-Shettigar) [reply app/src/main/java/org/oppia/android/app/testing/MathExpressionInteractionsViewTestActivity.kt]: Removed.

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (Vishwajith-Shettigar) [reply app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: Done.

**Comment by Vishwajith-Shettigar** [comment]
Thank you @theMr17 for reviewing, assigning back to you, PTAL.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (theMr17) [inline app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt]: Nit: This empty line is still left out, please remove it. Line 156.

**Review by theMr17 — APPROVED** [review]
Thanks @Vishwajith-Shettigar, this LGTM! I just had one minor comment inline. Also, please add an explanation to the PR description regarding the changes in the [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) folder as per PR Essential Checklist.

**Comment by theMr17** [comment]
> @theMr17, could you please take a review pass on this PR? I had earlier requested some changes, please verify that they have been addressed, and please assign back to me once there is no comment to be addressed from your side.

@adhiamboperes PTAL. I have reviewed the PR and this LGTM now. I have also verified that your earlier requested changes have been addressed.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (theMr17) [reply app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt]: This has been resolved in the latest changes.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (theMr17) [reply app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt]: Resolved in the latest changes.

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (theMr17) [reply app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: Resolved in the latest changes.

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (theMr17) [reply app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: Resolved in the latest changes.

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (theMr17) [reply app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: Resolved in the latest changes.

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (theMr17) [reply app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: Resolved in latest changes.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (theMr17) [reply app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt:1]: Minor comment pending related to this.

**Comment by Vishwajith-Shettigar** [comment]
@adhiamboperes, PTAL.

File: app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt
Comment (Vishwajith-Shettigar) [reply app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/MathExpressionInteractionsViewModel.kt]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (adhiamboperes) [inline app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: This could be simplified to -> because click actions will only happen if the view is at least 90% displayed, so it is kind of a redundant check. Same applies to the other tests below.
```suggestion
      onView(withId(R.id.submit_button)).perform(click())
```

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @Vishwajith-Shettigar! Just had one more comment. 

Thanks for reviewing @theMr17!

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (Vishwajith-Shettigar) [reply app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: Done.

**Review by adhiamboperes — APPROVED** [review]
Thanks @Vishwajith-Shettigar, This LGTM. 

Assigning @BenHenning for codeowners

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: Please revert these style-only changes.

File: app/src/main/res/values/strings.xml
Comment (BenHenning) [inline app/src/main/res/values/strings.xml]: Should be ``math_equation_error_empty_input`` for parity.

File: app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt]: Ditto here, please revert style-only changes.

File: app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt
Comment (BenHenning) [reply app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt:3705]: Could you update the tests to verify that the button is enabled, instead? I suggest swapping all the previous removed lines with:

```kotlin
verifySubmitAnswerButtonIsEnabled() // Wrong answers shouldn't disable submit.
```

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @Vishwajith-Shettigar! Took a pass and had just a few comments. PTAL.

Also, could you update the branch to the latest develop (as there are conflicts to resolve)?

File: app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt
Comment (Vishwajith-Shettigar) [reply app/src/sharedTest/java/org/oppia/android/app/customview/interaction/MathExpressionInteractionsViewTest.kt]: okay.

File: app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt
Comment (Vishwajith-Shettigar) [reply app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt]: Done.

File: app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt
Comment (Vishwajith-Shettigar) [reply app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt:3705]: I will add, thank you.

File: app/src/main/res/values/strings.xml
Comment (Vishwajith-Shettigar) [reply app/src/main/res/values/strings.xml]: I will modify, thank you.

**Comment by Vishwajith-Shettigar** [comment]
Thank you @BenHenning, PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt:3289]: This & the removal on line 3470 also need an update & comment for parity with the other removals of the call to this method.

**Review by BenHenning — COMMENTED** [review]
Thanks @Vishwajith-Shettigar! Just one follow-up--PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt
Comment (Vishwajith-Shettigar) [reply app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt:3289]: I apologize for the mistake, it has now been updated. Thank you.

**Comment by Vishwajith-Shettigar** [comment]
@BenHenning, PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt
Comment (BenHenning) [reply app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt:3289]: No worries--thanks for fixing it!

**Review by BenHenning — APPROVED** [review]
Thanks @Vishwajith-Shettigar! This LGTM.

**Pull request merged**
